File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/tot/Checker.java
package com.xebia.functional.xef.java.auto.tot;

public class Checker {

    public static <A> Solutions.Solution<A> checkSolution(Solutions.Solution<A> response){
        System.out.println("✅ Validating solution: " + Rendering.truncateText(response.answer) + "...");
        return response.isValid ? response :
                Solutions.makeSolution(response.answer, false, "Invalid solution", null);
    }
}

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/tot/Checker.kt
package com.xebia.functional.xef.auto.tot

// Function to perform solution validation
internal fun <A> checkSolution(response: Solution<A>): Solution<A> {
  println("✅ Validating solution: ${truncateText(response.answer)}...")
  return if (response.isValid) response
  else Solution(response.answer, false, "Invalid solution", null)
}
